src/hotspot/share/opto/loopnode.cpp

```diff
@@ -6145,6 +6175,11 @@ void PhaseIdealLoop::build_loop_late_post_work(Node *n, bool pinned) {
   IdealLoopTree *chosen_loop = get_loop(least);
   if( !chosen_loop->_child )   // Inner loop?
     chosen_loop->_body.push(n);// Collect inner loops
+
+  if (!_verify_only && n->Opcode() == Op_OpaqueZeroTripGuard) {
+    _zero_trip_guard_opaque_nodes.push(n);
```

Comment:
That's a good idea to collect them newly here for each loop opts pass.

Reply:
No action required. I was first trying to suggest to move it to `Compile` to the other predicate opaque node lists but then I've thought that this solution here is cleaner. So it was more of a :+1: (I will refactor these predicate lists in the assertion predicate changes - otherwise, I would have suggested to move them here as well at some point).

Reply:
Thanks for the clarification.